### ISSUE TITLE
Provide Haproxy with Lua support

### ISSUE THREAD

**Issue by fuzzy-id** [NixOS/nixpkgs#23806]
## Issue description
At the moment Haproxy is built without the support of external Lua scripts. I managed to build a Haproxy with Lua support with the following config:
```nix
nixpkgs.config.packageOverrides = pkgs : rec {
  haproxy = pkgs.haproxy.overrideDerivation (oldAttrs : {
    buildInputs = oldAttrs.buildInputs ++ [pkgs.lua5_3];
    preConfigure = oldAttrs.preConfigure + ''makeFlags+=" USE_LUA=yes LUA_LIB=${pkgs.lua5_3}/lib LUA_INC=${pkgs.lua5_3}/include"'';
  });
};
```

Is there interest in having Haproxy with Lua support in nixpkgs? If so, I would like to contribute this as it seems like a nice first-starter.

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)

16.09.1829.c88e67d (Flounder)

* Nix version: (run `nix-env --version`)

nix-env (Nix) 1.11.7

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

"16.09.1829.c88e67d"

**Comment by Mic92** [comment]
I have no concerns and it seems to be also adopted by other downstream distributions.

**Comment by 0xABAB** [comment]
The solution I would like to see in general is to by default include all dependencies (or alternatively those used by popular distributions), but allow flags to be passed to disable features. 

This way it's possible to first get something working and then in production disable unneeded features.

**Comment by fuzzy-id** [comment]
Implemented by #24380.
